plugins/by-name/gitlab/default.nix

```diff
@@ -0,0 +1,137 @@
+{
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib.nixvim) defaultNullOpts;
+  inherit (lib) types;
+in
+lib.nixvim.neovim-plugin.mkNeovimPlugin {
+  name = "gitlab";
+  originalName = "gitlab.vim";
+  package = "gitlab-vim";
+
+  maintainers = [ lib.maintainers.GaetanLepage ];
+
+  extraOptions = {
+    nodePackage = lib.mkPackageOption pkgs "nodejs" {
+      nullable = true;
+      default = null;
+    };
+  };
+
+  extraConfig = cfg: {
+    extraPackages = [ cfg.nodePackage ];
+  };
+
+  settingsOptions = {
+    gitlab_url = defaultNullOpts.mkStr "https://gitlab.com" ''
+      The GitLab instance url to use if not `https://gitlab.com`.
+    '';
+
+    statuslines = {
+      enable = defaultNullOpts.mkBool true ''
+        Whether to enable statuslines.
+      '';
+    };
+
+    resource_editing = {
+      enable = defaultNullOpts.mkBool false ''
+        Whether to enable resource editing.
+      '';
+    };
+
+    minimal_message_level = defaultNullOpts.mkUnsignedInt 0 ''
+      Minimal message level for logs.
+    '';
+
+    code_suggestions = {
+      auto_filetypes =
+        defaultNullOpts.mkListOf types.str
+          [
+            "c"
+            "cpp"
+            "csharp"
+            "go"
+            "java"
+            "javascript"
+            "javascriptreact"
+            "kotlin"
+            "markdown"
+            "objective-c"
+            "objective-cpp"
+            "php"
+            "python"
+            "ruby"
+            "rust"
+            "scala"
+            "sql"
+            "swift"
+            "terraform"
+            "typescript"
+            "typescriptreact"
+            "sh"
+            "html"
+            "css"
+          ]
+          ''
+            Filetypes to automatically invoke `|gitlab.code_suggestions.start()|`.
+          '';
+
+      enabled = defaultNullOpts.mkBool true ''
+        Whether to enable `|gitlab-code-suggestions|` via the LSP binary.
+      '';
+
+      fix_newlines = defaultNullOpts.mkBool true ''
+        Whether to replace newlines that have become null-byte due to switching between encodings.
+      '';
+
+      lsp_binary_path = defaultNullOpts.mkStr' {
+        pluginDefault = "node";
+        example = lib.literalExpression "lib.getExe pkgs.nodejs";
+        description = ''
+          The path where the language server binary is available or should be installed to.
```

Comment:
Q: the option relates to node, but the description is talking about a lsp server?

Reply:
Seems it relates to the runtime that will run the LSP (nodejs), not the LSP itself (gitlab-lsp)

Reply:
Rephrased !